yew/src/virtual_dom/mod.rs

```diff
@@ -47,18 +47,108 @@ pub use self::vtext::VText;
 pub trait Listener {
     /// Returns the name of the event
     fn kind(&self) -> &'static str;
-    /// Attaches a listener to the element.
+
+    /// Attaches a listener to the Element
+    #[cfg(feature = "std_web")]
     fn attach(&self, element: &Element) -> EventListener;
+
+    /// Handles an event firing
+    #[cfg(feature = "web_sys")]
+    fn handle(&self, event: Event);
+
+    /// Defines flags toi modify the handling of the event. See yew::callback for more details.
+    #[cfg(feature = "web_sys")]
+    fn flags(&self) -> u8;
 }
 
 impl fmt::Debug for dyn Listener {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "Listener {{ kind: {} }}", self.kind())
+        cfg_match! {
+            feature = "web_sys" => write!(
+                f,
+                "Listener {{ kind: {}, flags: {:#X} }}",
+                self.kind(),
+                self.flags()
+            ),
+            feature = "stdweb" => write!(
+                f,
+                "Listener {{ kind: {} }}",
+                self.kind()
+            ),
+        }
+    }
+}
+
+/// A list of event listeners
+#[derive(Debug, Clone)]
+pub enum Listeners {
+    /// Added to global registry by ID
+    #[cfg(feature = "web_sys")]
+    Registered(u64),
+
+    /// Added to the Element. Stored so they are removed on VTag drop.
+    #[cfg(feature = "std_web")]
+    Registered(Rc<Vec<EventListener>>),
+
+    /// Not yet added to the element
+    Pending(Vec<Rc<dyn Listener>>),
+}
+
+impl Default for Listeners {
+    fn default() -> Self {
+        Self::Pending(Default::default())
```

Comment:
Shorter and more descriptive:
```suggestion
        Self::Pending(Vec::new())
```